src/precice/impl/ParticipantImpl.cpp

```diff
@@ -1609,6 +1849,8 @@ void ParticipantImpl::resetWrittenData()
 {
   PRECICE_TRACE();
   for (auto &context : _accessor->writeDataContexts()) {
+    // reset the buffered data here
+    context.invalidateMappingCache(true);
     context.resetBuffer();
```

Comment:
```suggestion
    // reset the buffered data here
    context.resetBufferedData()
```

and create function inside `WriteDataContext`:

```
void resetBufferedData() {
    invalidateMappingCache(true);
    resetBuffer();  // <- make private or even inline?
}
```

Reply:
put it inline